packages/@o3r/extractors/src/core/wrapper.ts

```diff
@@ -8,10 +13,61 @@ const noopBuilderWrapper: BuilderWrapper = (fn) => fn;
  * @param builderFn
  */
 export const createBuilderWithMetricsIfInstalled: BuilderWrapper = (builderFn) => async (opts, ctx) => {
+  const angularJsonPath = path.join(ctx.workspaceRoot, 'angular.json');
+  const nxJsonPath = path.join(ctx.workspaceRoot, 'nx.json');
+  const configPath = existsSync(angularJsonPath)
+    ? angularJsonPath
+    : existsSync(nxJsonPath)
+      ? nxJsonPath
+      : '';
+  const configJson = JSON.parse(
+    existsSync(configPath)
+      ? readFileSync(configPath, {encoding: 'utf8'}).toString()
+      : '{}'
+  );
   let wrapper: BuilderWrapper = noopBuilderWrapper;
   try {
     const { createBuilderWithMetrics } = await import('@o3r/telemetry');
-    wrapper = createBuilderWithMetrics;
-  } catch {}
+    if (configJson.cli?.o3rMetrics) {
+      wrapper = createBuilderWithMetrics;
+    }
+  } catch (e: any) {
+    // Do not throw if `@o3r/telemetry is not installed
+    if (configJson.cli?.o3rMetrics !== false) {
```

Comment:
so you will ask the question in case `configJson.cli.o3rMetrics === true` and the package is not installed?
Maybe more better to
```suggestion
    if (typeof configJson.cli?.o3rMetrics === 'undefined') {
```

and display a warning in case  `configJson.cli.o3rMetrics === true` to request the installation